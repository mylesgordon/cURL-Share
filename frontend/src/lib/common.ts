export function splitAndTrim(input: string): Array<string> {
		return input.split(',').map((item) => item.trim());
	}