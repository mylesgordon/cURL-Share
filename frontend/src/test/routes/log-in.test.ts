import LogIn from '../../routes/log-in/+page.svelte';
import { render } from '@testing-library/svelte';
import { testMatchingSnapshot } from '../common';

describe('Log In page', () => {
	testMatchingSnapshot(LogIn)

	it('should display the correct title', () => {
		render(LogIn);
		expect(document.title).toEqual('Log In | cURL Share');
	});


});