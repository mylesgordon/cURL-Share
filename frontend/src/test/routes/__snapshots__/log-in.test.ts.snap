// Vitest Snapshot v1

exports[`Index page > should match snapshot 1`] = `
<body>
  <div>
    <!--&lt;Meta&gt;-->
     
    <form
      class="flex flex-col space-y-3"
    >
      <div
        class="w-100 s-JZthjkXlZrq3"
      >
        <label
          class="label s-JZthjkXlZrq3"
          for="username"
        >
          Username
        </label>
         
        <input
          class="input input-rounded s-JZthjkXlZrq3"
          id="username"
          type="text"
        />
         
      </div>
      <!--&lt;Input&gt;-->
       
      <div
        class="w-100 s-JZthjkXlZrq3"
      >
        <label
          class="label s-JZthjkXlZrq3"
          for="password"
        >
          Password
        </label>
         
        <input
          class="input input-rounded s-JZthjkXlZrq3"
          id="password"
          type="password"
        />
         
      </div>
      <!--&lt;Input&gt;-->
       
      <div
        aria-label="Form submission buttons"
        class="btn-group self-center s-_5WuMKolmwWJ"
        role="group"
      >
        <button
          class="btn btn-primary btn-bordered btn-grouped s-NAQuwh5ehqbV"
          type="submit"
        >
          Log In
        </button>
        
        <!--&lt;Button&gt;-->
         
        <button
          class="btn btn-primary btn-bordered btn-grouped s-NAQuwh5ehqbV"
          type="button"
        >
          Sign Up
        </button>
        
        <!--&lt;Button&gt;-->
      </div>
      <!--&lt;ButtonGroup&gt;-->
    </form>
    <!--&lt;+page&gt;-->
  </div>
</body>
`;

exports[`Log In page > should match snapshot 1`] = `
<body>
  <div>
    <!--&lt;Meta&gt;-->
     
    <form
      class="flex flex-col space-y-3"
    >
      <div
        class="w-100 s-JZthjkXlZrq3"
      >
        <label
          class="label s-JZthjkXlZrq3"
          for="username"
        >
          Username
        </label>
         
        <input
          class="input input-rounded s-JZthjkXlZrq3"
          id="username"
          type="text"
        />
         
      </div>
      <!--&lt;Input&gt;-->
       
      <div
        class="w-100 s-JZthjkXlZrq3"
      >
        <label
          class="label s-JZthjkXlZrq3"
          for="password"
        >
          Password
        </label>
         
        <input
          class="input input-rounded s-JZthjkXlZrq3"
          id="password"
          type="password"
        />
         
      </div>
      <!--&lt;Input&gt;-->
       
      <div
        aria-label="Form submission buttons"
        class="btn-group self-center s-_5WuMKolmwWJ"
        role="group"
      >
        <button
          class="btn btn-primary btn-bordered btn-grouped s-NAQuwh5ehqbV"
          type="submit"
        >
          Log In
        </button>
        
        <!--&lt;Button&gt;-->
         
        <button
          class="btn btn-primary btn-bordered btn-grouped s-NAQuwh5ehqbV"
          type="button"
        >
          Sign Up
        </button>
        
        <!--&lt;Button&gt;-->
      </div>
      <!--&lt;ButtonGroup&gt;-->
    </form>
    <!--&lt;+page&gt;-->
  </div>
</body>
`;
